Actually use mongoose model to resolve queries

NeoTheThird opened the pull request:
This is a part of #27. Thoughts appreciated. To fully close that issue, i propose allowing the setting default fields to [populate](https://mongoosejs.com/docs/populate.html) in the constructor. I'd be happy to take a swing at that as well, if there's interest.

NeoTheThird commented:
Actually, i'm now thinking it might be best to recommend [mongoose-autopopulate](https://www.npmjs.com/package/mongoose-autopopulate) for that. This way no further code is needed in this lib and we could consider #27 closed with this. I'd be happy to write a note for the readme.

lorensr commented:
Thanks Neo! Do you have a take on the implementation in #30? And sounds good re: Readme note, thank you ☺️

NeoTheThird commented:
I wasn't aware of exec, added it to my PR as well and added that readme note. I think it's a little more elegant to pass in the model directly rather than checking isModel again like #30 does, but i guess that comes down to personal preference. Go with whatever PR you like :) the external api would remain untouched anyways, so it should be up to you, the maintainer, not us, the users of the lib.

NeoTheThird commented on README.md:
```diff
@@ -144,8 +146,8 @@ class Users extends MongoDataSource {
   updateUserName(userId, newName) {
     this.deleteFromCacheById(userId)
-    return this.collection.updateOne({ 
-      _id: userId 
```
only saw this after my commit, i got my editor set up to remove trailing spaces automagically. unrelated ofc, but i guess they technically have no business being there ;)

lorensr commented:
Thanks a lot, I combined them. Published in `0.2.7`
